src/binary-reader.cc

```diff
@@ -176,7 +176,7 @@ class BinaryReader {
   Index NumTotalMemories();
   Index NumTotalGlobals();
 
-  Result ReadInitExpr(Index index) WABT_WARN_UNUSED;
+  Result ReadInitExpr(Index index, bool require_i32 = false) WABT_WARN_UNUSED;
```

Comment:
Might be nice to have a passthrough function called `ReadI32InitExpr` that sets this to `true`, especially if we add the same thing for elem segments too.

Reply:
Done.  PTAL